gas/config/tc-riscv.c

```diff
@@ -121,15 +121,56 @@ riscv_subset_supports (const char *feature)
 }
 
 static bfd_boolean
-riscv_multi_subset_supports (const char *features[])
+riscv_multi_subset_supports (enum riscv_insn_class insn_class)
 {
-  unsigned i = 0;
-  bfd_boolean supported = TRUE;
+  switch (insn_class)
+    {
+    case INSN_CLASS_I: return riscv_subset_supports ("i");
+    case INSN_CLASS_C: return riscv_subset_supports ("c");
+    case INSN_CLASS_A: return riscv_subset_supports ("a");
+    case INSN_CLASS_M: return riscv_subset_supports ("m");
+    case INSN_CLASS_F: return riscv_subset_supports ("f");
+
+    case INSN_CLASS_D_C:
+      return riscv_subset_supports ("d") && riscv_subset_supports ("c");
+    case INSN_CLASS_F_C: return riscv_subset_supports ("i");
+      return riscv_subset_supports ("f") && riscv_subset_supports ("c");
+
+    case INSN_CLASS_Q: return riscv_subset_supports ("q");
+
+    case INSN_CLASS_B: return riscv_subset_supports ("b");
+    case INSN_CLASS_B_ZBB:
+      return riscv_subset_supports ("b") || riscv_subset_supports ("zbb");
+
+    case INSN_CLASS_B_ZBC:
+      return riscv_subset_supports ("b") || riscv_subset_supports ("zbc");
+
+    case INSN_CLASS_B_ZBE:
+      return riscv_subset_supports ("b") || riscv_subset_supports ("zbe");
+
+    case INSN_CLASS_B_ZBM:
+      return riscv_subset_supports ("b") || riscv_subset_supports ("zbm");
```

Comment:
`zbr`, `zbm` and `zbt` not included by `b`? it should just `return riscv_subset_supports ("zbm");`

Reply:
I also forgot about ZBF, which I added in latest push. I fixed the other classes too (I just copy pasted them, and forgot to amend.)

I think they are all classified correctly now.